include/deal.II/lac/precondition.h

```diff
@@ -1937,6 +1937,40 @@ namespace internal
         }
     }
 
+    template <typename VectorType>
+    void set_initial_guess(VectorType &vector)
+    {
+      vector = 1./std::sqrt(static_cast<double>(vector.size()));
+      if (vector.locally_owned_elements().is_element(0))
+        vector(0) = 0.;
+    }
+
+    template <typename Number>
+    void set_initial_guess(::dealii::Vector<Number> &vector)
+    {
+      // Choose a high-frequency mode consisting of numbers between 0 and 1
+      // that is cheap to compute (cheaper than random numbers) but avoids
+      // obviously re-occurring numbers in multi-component systems by choosing
+      // a period of 11
+      const unsigned int size = vector.size();
+      for (unsigned int i=0; i<size; )
+        for (unsigned int j=0; j<11 && i<size; ++j, ++i)
+          vector(i) = j;
```

Comment:
I had a short look at the assembler code and it appears that the compiler is able to avoid the integer division. So definitely this is not be visible anywhere and I appreciate the code cleanliness.